### PR TITLE
AssetForm: select institution via select box

### LINKED ISSUE #56: Lookup institution component
Use this information to show a list of institutions the user belong to in the select when you create a new asset

### PR DESCRIPTION
**Pull request by rjw57**
Instead of using a free text field, use a select box to render the institution/department field of the form. By design this doesn't work if the asset is from another institution since the select box is only populated with the user's own institutions.

Closes #56.

### CODE REVIEW

File: src/components/SelectInstitution.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
```
Comment (msb) [inline src/components/SelectInstitution.js]: remove this as not used

File: src/components/SelectInstitution.test.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
```
Comment (msb) [inline src/components/SelectInstitution.test.js]: as above

File: src/components/OwnInstitutionField.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import InstitutionField from './InstitutionField';
+
+const mapStateToProps = ({ lookupApi: { self } }, { extraInstids = [] }) => ({
+  instids: [
+    ...(self ? self.institutions.map(({ instid }) => instid) : []),
+    ...extraInstids,
```
Comment (msb) [inline src/components/OwnInstitutionField.js]: extraInstids not required at the moment

File: src/components/InstitutionField.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        <MenuItem key={index} value={value}>{ label }</MenuItem>
+      ))
+    ]
+  }
+  </TextField>
+);
+
+InstitutionField.propTypes = {
+  institutionValuesAndLabels: PropTypes.arrayOf(PropTypes.object).isRequired,
+  allowNone: PropTypes.bool,
```
Comment (msb) [inline src/components/InstitutionField.js]: would be a good idea to add instids to propTypes

**Comment by msb** [comment]
the 1st department isn't pre-selected in the create form

**Comment by rjw57** [comment]
Should it be?

**Comment by rjw57** [comment]
To be less flippant: the 1st department is no likely or less to be the one the user is creating an entry for. Indeed, leaving it pre-selected almost guarantees the user will forget to change it at some point.

I suppose it could be pre-selected if the user has only one institution...

**Comment by joeadams04** [comment]
I agree it should be pre-selected if the user only has one institution. You could even disable it as well if we wanted, as it's impossible to add an entry on behalf of another institution.

I also agree that pre-selecting when there are two or more institutions is probably a bad idea, as some user's are bound to miss it. It will obviously be one more field for them to complete, but should reduce the number of errors significantly.

**Pull request merged**